gatewayconfig/constants.py

```diff
@@ -81,7 +81,7 @@
     "100": "connected",  # connected
     "50": "already",  # Already Connecting
     "60": "invalid",  # Invalid Key
-    "30": "failed"   # Connection Failed
+    "30": "invalid"   # Incorrect Password
```

Comment:
Do you know where these numbers come from? Will changing the value here have an impact for other situations (not incorrect password) that also use a `30` status?

Reply:
```
device_states = { 0: "Unknown",
                 10: "Unmanaged",
                 20: "Unavailable",
                 30: "Disconnected",
                 40: "Prepare",
                 50: "Config",
                 60: "Need Auth",
                 70: "IP Config",
                 80: "IP Check",
                 90: "Secondaries",
                 100: "Activated",
                 110: "Deactivating",
                 120: "Failed" }
```
https://github.com/lcp/NetworkManager/blob/master/examples/python/nm-state.py
https://cgit.freedesktop.org/NetworkManager/NetworkManager/tree/examples/python/dbus/nm-state.py
https://people.freedesktop.org/~lkundrak/nm-docs/nm-dbus-types.html
https://developer-old.gnome.org/NetworkManager/stable/nmcli-examples.html

Reply:
Rather than above dictionary mapping, the following enum constants seems to be more convenient for checking the wifi state.
https://github.com/ProtoThis/NetworkManager/blob/master/tools/test-networkmanager-service.py#L28
```
# Device state
NM_DEVICE_STATE_UNKNOWN      = 0
NM_DEVICE_STATE_UNMANAGED    = 10
NM_DEVICE_STATE_UNAVAILABLE  = 20
NM_DEVICE_STATE_DISCONNECTED = 30
NM_DEVICE_STATE_PREPARE      = 40
NM_DEVICE_STATE_CONFIG       = 50
NM_DEVICE_STATE_NEED_AUTH    = 60
NM_DEVICE_STATE_IP_CONFIG    = 70
NM_DEVICE_STATE_IP_CHECK     = 80
NM_DEVICE_STATE_SECONDARIES  = 90
NM_DEVICE_STATE_ACTIVATED    = 100
NM_DEVICE_STATE_DEACTIVATING = 110
NM_DEVICE_STATE_FAILED       = 120
```

Reply:
Thanks for finding this @posterzh, very helpful. My concern is still that other failure modes will also return 30 and that we will now treat all of them as an incorrect password. The error message is being in the app right? Do you have a link to that code?